### PR TITLE
[match/cert/sigh] support xcode 11 profiles

### LINKED ISSUE #15031: [match] Generate profiles with new Xcode 11 certificates attached
#### Motivation Behind Feature
To be able to use Xcode 11 Beta on devices and in CI.

#### Feature Description
Now Xcode 11 has some new kind of development Certificates. Actually there are 2 kinds of development certificates now. One for Xcode 10.x and below and one for Xcode 11.x and above. The provisioning profiles generated by `match` are only connected to the Xcode 10x  cert. 

It would be nice if every profile generated would support both certs.

### LINKED ISSUE #15214: fastlane sigh action does not include apple cert when regenerating development profiles
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
When using fastlane sigh to regenerate development profiles, it will remove all apple certificates and only include iOS certificates on the new development profiles

eg.
`fastlane sigh --development -a com.company.App -u myemail@company.com --team_id XXXXXXXXX --skip_certificate_verification --skip_install --provisioning_name='Universal iOS Development' --force --template_name 'Special Service (Dev)' -q "Universal_iOS_Development.mobileprovision" -o provision`

will create a development profile that only include all the account iOS certs, and leaving all apple certs out of the new profile.

##### Command executed
`fastlane sigh --development -a com.company.App -u myemail@company.com --team_id XXXXXXXXX --skip_certificate_verification --skip_install --provisioning_name='Universal iOS Development' --force --template_name 'Special Service (Dev)' -q "Universal_iOS_Development.mobileprovision" -o provision`

##### Complete output when running fastlane, including the stack trace and command used

```
[21:23:41]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile

+-------------------------------------+--------------------------------------+
|                          Summary for sigh 2.129.0                          |
+-------------------------------------+--------------------------------------+
| development                         | true                                 |
| app_identifier                      | com.company.App                  |
| username                            | myemail@company.com                   |
| team_id                             | XXXXXXXXX                           |
| skip_certificate_verification       | true                                 |
| skip_install                        | true                                 |
| provisioning_name                   | Universal iOS Development            |
| force                               | true                                 |
| template_name                       |Core Service (Dev)       |
| filename                            | Universal_iOS_Development.mobilepro  |
|                                     | vision                               |
| adhoc                               | false                                |
| developer_id                        | false                                |
| ignore_profiles_with_different_name | false                                |
| skip_fetch_profiles                 | false                                |
| platform                            | ios                                  |
| readonly                            | false                                |
+-------------------------------------+--------------------------------------+

[21:23:43]: Starting login with user 'myemail@company.com'
[21:23:44]: Successfully logged in
[21:23:44]: Fetching profiles...
[21:23:44]: Found 1 matching profile(s)
[21:23:44]: Recreating the profile
[21:23:45]: Creating new provisioning profile for 'com.company.App' with name 'Universal iOS Development' for 'ios' platform
[21:23:46]: Downloading provisioning profile...
[21:23:46]: Successfully downloaded provisioning profile...
/Users/myname/Universal_iOS_Development.mobileprovision
```

### Environment

<details>
  <pre> <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.6                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.17.2 (Apple Git-113)          |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.6 (17G3025)                  |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 10.1                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.129.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.8.1        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.1.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.9.5        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| naturally                 | 2.1.0        |
| simctl                    | 1.6.5        |
| jwt                       | 2.1.0        |
| uber                      | 0.0.15       |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| multi_json                | 1.12.1       |
| signet                    | 0.11.0       |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.2.7        |
| digest-crc                | 0.4.1        |
| google-cloud-storage      | 1.15.0       |
| emoji_regex               | 0.1.1        |
| libxml-ruby               | 3.0.0        |
| psych                     | 2.0.8        |
</details>

*generated on:* **2019-08-21**
</details>

### LINKED ISSUE #15300: sigh is not working with the new `apple developer` / `apple distribution` certificates introduced with Xcode 11
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
* On of my lanes builds an ad-hoc version of my company app, for this we use sigh to manage generate provisioning profiles, until now it has been working fine but on our Xcode 11 branch we noticed the `iOS distribution` certificate we were using was not available on Xcode 11 and the only option to generate new ones is `apple development` and `apple distribution`.

* After generating an `apple distribution` certificate and updating our project to use that certificate we found out that sigh always fails to generate provisioning profiles and complains about "not find a matching code signing identity for type 'AdHoc'"

* We also tried re-generating an "old" `ios distribution` certificate and try to run our lane but this time while `sigh` was able to generate provisioning profiles but then fails when code signing with error `error: exportArchive: Provisioning profile "com.hellofresh.HelloFresh.mainwidget AdHoc" doesn't include signing certificate "Apple Distribution: Hellofresh GmbH (2A7X7KDSR2)".`

(ps: I am aware we could use match, this is not what this issue is about)

##### Command executed
`bundle exec fastlane upload_fabric configuration:Fabric`

##### Complete output when running fastlane, including the stack trace and command used

running on Xcode 10.3 (works fine as expected):
```
[15:35:07]: [0;32;49m------------------[0m
[15:35:07]: [0;32;49m--- Step: sigh ---[0m
[15:35:07]: [0;32;49m------------------[0m

+-------------------------------------+---------------------------+
|                    [0;32;49mSummary for sigh 2.130.0[0m                     |
+-------------------------------------+---------------------------+
| username                            | xxx     |
| app_identifier                      | com.hellofresh.HelloFresh |
| adhoc                               | true                      |
| force                               | true                      |
| developer_id                        | false                     |
| development                         | false                     |
| skip_install                        | false                     |
| team_id                             | xxx                |
| ignore_profiles_with_different_name | false                     |
| skip_fetch_profiles                 | false                     |
| skip_certificate_verification       | false                     |
| platform                            | ios                       |
| readonly                            | false                     |
+-------------------------------------+---------------------------+

[15:35:07]: Starting login with user 'xxx'
[15:35:08]: Successfully logged in
[15:35:08]: Fetching profiles...
[15:35:09]: Verifying certificates...
[15:35:11]: [0;32;49mFound 1 matching profile(s)[0m
[15:35:11]: [0;33;49mRecreating the profile[0m
[15:35:15]: [0;33;49mCreating new provisioning profile for 'com.hellofresh.HelloFresh' with name 'com.hellofresh.HelloFresh AdHoc' for 'ios' platform[0m
[15:35:17]: [0;33;49mDownloading provisioning profile...[0m
[15:35:17]: [0;32;49mSuccessfully downloaded provisioning profile...[0m
[15:35:18]: Installing provisioning profile...
[0;32;49m/Users/device.de_ci/.jenkins/workspace/Upload_Fabric/HelloFresh/fastlane/provisioning_profiles/AdHoc_com.hellofresh.HelloFresh.mobileprovision[0m
[15:35:18]: Setting Provisioning Profile type to 'ad-hoc'
```

running on Xcode 11 (does not work):
```
[15:23:30]: [0;32;49m------------------[0m
[15:23:30]: [0;32;49m--- Step: sigh ---[0m
[15:23:30]: [0;32;49m------------------[0m

+-------------------------------------+---------------------------+
|                    [0;32;49mSummary for sigh 2.130.0[0m                     |
+-------------------------------------+---------------------------+
| username                            | xxx     |
| app_identifier                      | com.hellofresh.HelloFresh |
| adhoc                               | true                      |
| force                               | true                      |
| developer_id                        | false                     |
| development                         | false                     |
| skip_install                        | false                     |
| team_id                             | xxx                |
| ignore_profiles_with_different_name | false                     |
| skip_fetch_profiles                 | false                     |
| skip_certificate_verification       | false                     |
| platform                            | ios                       |
| readonly                            | false                     |
+-------------------------------------+---------------------------+

[15:23:30]: Starting login with user 'xxx'
[15:23:31]: Successfully logged in
[15:23:31]: Fetching profiles...
[15:23:32]: Verifying certificates...
[15:23:34]: [0;32;49mFound 1 matching profile(s)[0m
[15:23:34]: [0;33;49mRecreating the profile[0m
[15:23:37]: [0;32;49m-----------------------------------[0m
[15:23:37]: [0;32;49m--- Step: clean_build_artifacts ---[0m
[15:23:37]: [0;32;49m-----------------------------------[0m
[15:23:37]: [0;32;49mCleaned up build artifacts 🐙[0m
+--------------------------------+-------------------+
|                    [0;33;49mLane Context[0m                    |
+--------------------------------+-------------------+
| DEFAULT_PLATFORM               | ios               |
| PLATFORM_NAME                  | ios               |
| LANE_NAME                      | ios upload_fabric |
| VERSION_NUMBER                 | 3.25              |
| LATEST_BUILD_NUMBER            | 10388             |
| LATEST_TESTFLIGHT_BUILD_NUMBER | 10388             |
| BUILD_NUMBER                   | 10391             |
+--------------------------------+-------------------+
[15:23:37]: [0;31;49mCould not find a matching code signing identity for type 'AdHoc'. It is recommended to use match to manage code signing for you, more information on https://codesigning.guide. If you don't want to do so, you can also use cert to generate a new one: https://fastlane.tools/cert[0m
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.6                                     |
| Ruby                        | 2.6.1                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.18.0                          |
| Installation Source         | /usr/local/lib/ruby/gems/2.6.0/bin/fastlane |
| Host                        | Mac OS X 10.14.6 (18G95)                    |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.6.1/lib            |
| OpenSSL Version             | OpenSSL 1.0.2q  20 Nov 2018                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.0                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.131.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                      | Version | Update-Status |
| --------------------------- | ------- | ------------- |
| fastlane-plugin-test_center | 3.8.3   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                         | Version      |
| --------------------------- | ------------ |
| did_you_mean                | 1.3.0        |
| bundler                     | 2.0.1        |
| CFPropertyList              | 3.0.1        |
| public_suffix               | 2.0.5        |
| addressable                 | 2.7.0        |
| atomos                      | 0.1.3        |
| babosa                      | 1.0.2        |
| claide                      | 1.0.3        |
| colored                     | 1.2          |
| colored2                    | 3.1.2        |
| colorize                    | 0.8.1        |
| highline                    | 1.7.10       |
| commander-fastlane          | 4.4.6        |
| declarative                 | 0.0.10       |
| declarative-option          | 0.1.0        |
| digest-crc                  | 0.4.1        |
| unf_ext                     | 0.0.7.6      |
| unf                         | 0.1.4        |
| domain_name                 | 0.5.20190701 |
| dotenv                      | 2.7.5        |
| emoji_regex                 | 1.0.1        |
| excon                       | 0.66.0       |
| multipart-post              | 2.0.0        |
| faraday                     | 0.15.4       |
| http-cookie                 | 1.0.3        |
| faraday-cookie_jar          | 0.0.6        |
| faraday_middleware          | 0.13.1       |
| fastimage                   | 2.1.7        |
| gh_inspector                | 1.1.3        |
| jwt                         | 2.1.0        |
| memoist                     | 0.16.0       |
| multi_json                  | 1.13.1       |
| os                          | 1.0.1        |
| signet                      | 0.11.0       |
| googleauth                  | 0.6.7        |
| httpclient                  | 2.8.3        |
| mime-types-data             | 3.2019.0904  |
| mime-types                  | 3.3          |
| uber                        | 0.1.0        |
| representable               | 3.0.4        |
| retriable                   | 3.1.2        |
| google-api-client           | 0.23.9       |
| google-cloud-env            | 1.2.1        |
| google-cloud-core           | 1.3.1        |
| google-cloud-storage        | 1.16.0       |
| json                        | 2.2.0        |
| mini_magick                 | 4.9.5        |
| multi_xml                   | 0.6.0        |
| plist                       | 3.5.0        |
| rubyzip                     | 1.2.4        |
| security                    | 0.1.3        |
| naturally                   | 2.2.0        |
| simctl                      | 1.6.6        |
| slack-notifier              | 2.3.2        |
| terminal-notifier           | 2.0.0        |
| unicode-display_width       | 1.6.0        |
| terminal-table              | 1.8.0        |
| tty-screen                  | 0.7.0        |
| tty-cursor                  | 0.7.0        |
| tty-spinner                 | 0.9.1        |
| word_wrap                   | 1.0.0        |
| nanaimo                     | 0.2.6        |
| xcodeproj                   | 1.12.0       |
| rouge                       | 2.0.7        |
| xcpretty                    | 0.3.0        |
| xcpretty-travis-formatter   | 1.0.0        |
| xctest_list                 | 1.1.8        |
| fastlane-plugin-test_center | 3.8.3        |
</details>

*generated on:* **2019-09-13**
</details>

### LINKED ISSUE #15369: Update Match to use the Xcode 11 compatible provisioning profiles
I've had to resort to manually uploading builds because Match doesn't support provisioning profiles for Xcode 11. This is the same issue as #15031 but it looks like literally nothing has been done. Can we get a fix for this asap please? This should be top priority as everyone is now upgrading to Xcode 11!

### LINKED ISSUE #15389: Unable to import new Apple Distribution private key to repo
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Due to Apple's new certificates from Xcode 11, I have been trying to manually encrypt updated provisioning profiles and certificates into our fastlane repo. However, whenever I export the Private Key as a .p12 for the new Apple Distribution certificate, it's not able to be read/downloaded anywhere (if I run match on another computer, the private key will not make it, just the cert)

##### Command executed

I've been following the steps here: https://docs.fastlane.tools/advanced/other/#manually-manage-the-fastlane-match-repo

The provisioning profile downloads and it tries to download the correct cert that I manually uploaded, but the private key doesn't come with it.

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.4                                     |
| Ruby                        | 2.5.1                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.21.0 (Apple Git-122)          |
| Installation Source         | ~/.rbenv/versions/2.5.1/bin/fastlane        |
| Host                        | Mac OS X 10.14.4 (18E226)                   |
| Ruby Lib Dir                | ~/.rbenv/versions/2.5.1/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2s  28 May 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.0                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.131.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                                | Version | Update-Status |
| ----------------------------------------------------- | ------- | ------------- |
| fastlane-plugin-badge                                 | 1.3.0   | ✅ Up-To-Date  |
| fastlane-plugin-aws_s3                                | 1.6.0   | ✅ Up-To-Date  |
| fastlane-plugin-versioning                            | 0.3.4   | ✅ Up-To-Date  |
| fastlane-plugin-get_unprovisioned_devices_from_hockey | 0.1.1   | ✅ Up-To-Date  |
| fastlane-plugin-rome                                  | 0.3.8   | ✅ Up-To-Date  |
| fastlane-plugin-sentry                                | 1.5.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                                   | Version      |
| ----------------------------------------------------- | ------------ |
| did_you_mean                                          | 1.2.0        |
| slack-notifier                                        | 2.3.2        |
| rouge                                                 | 2.0.7        |
| xcpretty                                              | 0.3.0        |
| terminal-notifier                                     | 2.0.0        |
| terminal-table                                        | 1.8.0        |
| multipart-post                                        | 2.0.0        |
| word_wrap                                             | 1.0.0        |
| public_suffix                                         | 2.0.5        |
| babosa                                                | 1.0.2        |
| colored                                               | 1.2          |
| highline                                              | 1.7.10       |
| commander-fastlane                                    | 4.4.6        |
| http-cookie                                           | 1.0.3        |
| faraday-cookie_jar                                    | 0.0.6        |
| gh_inspector                                          | 1.1.3        |
| mini_magick                                           | 4.9.5        |
| multi_xml                                             | 0.6.0        |
| security                                              | 0.1.3        |
| xcpretty-travis-formatter                             | 1.0.0        |
| jwt                                                   | 2.1.0        |
| uber                                                  | 0.1.0        |
| declarative                                           | 0.0.10       |
| declarative-option                                    | 0.1.0        |
| representable                                         | 3.0.4        |
| retriable                                             | 3.1.2        |
| multi_json                                            | 1.13.1       |
| signet                                                | 0.11.0       |
| memoist                                               | 0.16.0       |
| googleauth                                            | 0.6.7        |
| httpclient                                            | 2.8.3        |
| google-api-client                                     | 0.23.9       |
| nanaimo                                               | 0.2.6        |
| colored2                                              | 3.1.2        |
| claide                                                | 1.0.3        |
| CFPropertyList                                        | 3.0.1        |
| atomos                                                | 0.1.3        |
| xcodeproj                                             | 1.12.0       |
| unicode-display_width                                 | 1.6.0        |
| plist                                                 | 3.5.0        |
| addressable                                           | 2.7.0        |
| tty-screen                                            | 0.7.0        |
| tty-cursor                                            | 0.7.0        |
| tty-spinner                                           | 0.9.1        |
| excon                                                 | 0.66.0       |
| faraday                                               | 0.15.4       |
| unf_ext                                               | 0.0.7.6      |
| unf                                                   | 0.1.4        |
| domain_name                                           | 0.5.20190701 |
| fastimage                                             | 2.1.7        |
| json                                                  | 2.2.0        |
| rubyzip                                               | 1.2.4        |
| dotenv                                                | 2.7.5        |
| bundler                                               | 2.0.2        |
| faraday_middleware                                    | 0.13.1       |
| naturally                                             | 2.2.0        |
| simctl                                                | 1.6.6        |
| mime-types-data                                       | 3.2019.0904  |
| mime-types                                            | 3.3          |
| os                                                    | 1.0.1        |
| digest-crc                                            | 0.4.1        |
| google-cloud-env                                      | 1.2.1        |
| google-cloud-core                                     | 1.3.1        |
| google-cloud-storage                                  | 1.16.0       |
| emoji_regex                                           | 1.0.1        |
| openssl                                               | 2.1.0        |
| date                                                  | 1.0.0        |
| fileutils                                             | 1.0.2        |
| mini_portile2                                         | 2.4.0        |
| nokogiri                                              | 1.10.4       |
| ipaddr                                                | 1.2.0        |
| psych                                                 | 3.0.2        |
| badge                                                 | 0.11.0       |
| fastlane-plugin-badge                                 | 1.3.0        |
| apktools                                              | 0.7.2        |
| fastlane-plugin-aws_s3                                | 1.6.0        |
| fastlane-plugin-versioning                            | 0.3.4        |
| fastlane-plugin-get_unprovisioned_devices_from_hockey | 0.1.1        |
| fastlane-plugin-rome                                  | 0.3.8        |
| fastlane-plugin-sentry                                | 1.5.0        |
</details>

*generated on:* **2019-09-26**
</details>

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation and Context
- Fixes #15300
- Fixes #15389
- Fixes #15369
- Fixes #15214
- Fixes #15031
- Adds support for "Apple Developer" and "Apple Distribution" provision profiles to `sigh` (and `match`)

### Description

❗Enterprise accounts have access to "Apple Development" but not "Apple Distribution"

#### Sigh
- Concatenates with `Spaceship.certificate.apple_development.all` or `Spaceship.certificate.apple_distribution.all` in `Sigh::Runner` in the method for `certificates_for_profile_and_platform`. This method is used for determining provisioning profiles are available  for a specific platform and development type.
- Before this PR, a profile could not be generated with the new certificate types.

#### Cert
- Will create certs using `Spaceship.certificate.apple_development.all` or `Spaceship.certificate.apple_distribution.all` if `generate_apple_certs` param is `true. This param will default to `true` if Xcode 11 is being used

#### Match
- Will forward `generate_apple_certs` from _match_ params into _cert_ params with same Xcode 11 default value
- Also fixes `match nuke` to detect the new cert types to 💣 

#### Before this PR
```sh
$ fastlane sigh -a <BUNDLE_ID> --force -i <CERT_ID>
...
WARN [2019-09-29 16:32:14.76]: No certificates for filter: Certificate ID: '<CERT_ID>'
```

#### After this PR
```sh
$ fastlane sigh -a <BUNDLE_ID> --force -i <CERT_ID>
...
WARN [2019-09-29 16:35:04.41]: Creating new provisioning profile for '<BUNDLE_ID>' with name '<BUNDLE_ID>' for 'ios' platform
WARN [2019-09-29 16:35:05.21]: Downloading provisioning profile...
```

### CODE REVIEW

**Comment by janpio** [comment]
Test failures remind me so much of the [`faraday` foo](https://github.com/fastlane/fastlane/issues/15398), but I don't really see how this could be connected.

**Comment by joshdholtz** [comment]
@janpio I lied! This isn't done yet... I also want to add a thing to get `cert` working with these too on create 😇

**Comment by janpio** [comment]
Well, rebasing and passing tests probably helps for that as well ;) (Will check back later then)

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: cert/lib/cert/runner.rb [hunk 1]
```diff
@@ -134,16 +134,22 @@ def certificates
...
-      when 'macos'
-        cert_type = Spaceship.certificate.mac_app_distribution
-        cert_type = Spaceship.certificate.mac_development if Cert.config[:development]
-
+      # Check if apple certs (Xcode 11 and later) should be used
+      if Cert.config[:generate_apple_certs]
+        cert_type = Spaceship.certificate.apple_distribution
+        cert_type = Spaceship.certificate.in_house if Spaceship.client.in_house? # Enterprise doesn't use Apple Distribution
+        cert_type = Spaceship.certificate.apple_development if Cert.config[:development]
+      else
```
Comment (snatchev) [inline cert/lib/cert/runner.rb]: early return instead of `else`

File: sigh/lib/sigh/runner.rb [hunk 2]
```diff
@@ -155,27 +155,35 @@ def certificates_for_profile_and_platform
       case Sigh.config[:platform].to_s
       when 'ios', 'tvos'
         if profile_type == Spaceship.provisioning_profile.Development
-          certificates = Spaceship.certificate.development.all
+          certificates = Spaceship.certificate.development.all +
+                         Spaceship.certificate.apple_development.all
         elsif profile_type == Spaceship.provisioning_profile.InHouse
-          certificates = Spaceship.certificate.in_house.all
+          certificates = Spaceship.certificate.in_house.all +
+                         Spaceship.certificate.apple_distribution.all
```
Comment (snatchev) [inline sigh/lib/sigh/runner.rb]: remove.

**Review by snatchev — APPROVED** [review]

**Pull request merged**